lms/static/scripts/frontend_apps/services/content-info-fetcher.js

```diff
@@ -0,0 +1,76 @@
+import { apiCall, urlPath } from '../utils/api';
+
+/**
+ * @typedef {import('./client-rpc').ContentInfoConfig} ContentInfoConfig
+ */
+
+/**
+ * Identifies the metadata provider and item ID for the content being displayed.
+ *
+ * @typedef ContentId
+ * @prop {string} source
```

Comment:
I have merged the `ContentId` and `ContentBannerConfig` types to make it clearer where the data that the `ContentInfoFetcher` works with comes from. I have also revised the `source` string so that it is a union of known provider names, currently just "jstor". I have left in the runtime check in case the backend were to send something else, although this is now strictly redundant if you trust the types.